File: 0x08-python-more_classes/4-rectangle.py
#!/usr/bin/python3


"""class rectangle"""


class Rectangle:
    """defines rectangle"""
    def __init__(self, width=0, height=0):
        self.width = width
        self.height = height

    @property
    def width(self):
        return self.__width

    @property
    def height(self):
        return self.__height

    @width.setter
    def width(self, value):
        if not isinstance(value, int):
            raise TypeError("width must be an integer")
        elif value < 0:
            raise ValueError("width must be >= 0")
        else:
            self.__width = value

    @height.setter
    def height(self, value):
        if not isinstance(value, int):
            raise TypeError("height must be an integer")
        elif value < 0:
            raise ValueError("height must be >= 0")
        else:
            self.__height = value

    def area(self):
        return self.__width * self.__height

    def perimeter(self):
        if self.__width == 0 or self.__height == 0:
            return 0
        else:
            return (self.__width + self.__height) * 2

    def __str__(self):
        str = ""
        if self.__width == 0 or self.__height == 0:
            return str
        else:
            for i in range(self.__height):
                for x in range(self.__width):
                    str += '#'
                if i < self.__height - 1:
                    str += '\n'
            return str

    def __repr__(self):
        rect = "Rectangle(" + str(self.__width)
        rect += ", " + str(self.__height) + ")"
        return rect

File: 0x13-javascript_objects_scopes_closures/4-rectangle.js
#!/usr/bin/node
class Rectangle {
    constructor(w, h) {
	if (w > 0 && h > 0) {
	    this.width = w;
	    this.height = h;
	}
    }


    print() {
	if (this.width && this.height) {
	    for (let i = 0; i < this.height; i++) {
		console.log('X'.repeat(this.width));
	    }
	}
    }

    rotate() {
	if (this.width && this.height) {
	    let k = this.width;
	    this.width = this.height;
	    this.height = k;
	}
    }

    double() {
	if (this.width && this.height) {
	    this.width = this.width * 2;
	    this.height = this.height * 2;
	}
    }
}

module.exports = Rectangle;
